python_modules/dagster/dagster/_core/executor/multiprocess.py

```diff
@@ -244,6 +248,32 @@ def execute(
                                 yield event_or_none
                                 active_execution.handle_event(event_or_none)
 
+                        except ChildProcessHangOnShutdownException as e:
+                            serializable_error = serializable_error_info_from_exc_info(
+                                sys.exc_info()
+                            )
+                            step_context = plan_context.for_step(
+                                active_execution.get_step_by_key(key)
+                            )
+                            yield DagsterEvent.engine_event(
+                                step_context,
+                                (
+                                    f"Multiprocess executor: child process {e.pid} for step"
+                                    f" {key} failed to shut down cleanly after completion."
+                                ),
+                                EngineEventData.engine_error(serializable_error),
+                            )
+                            step_failure_event = DagsterEvent.step_failure_event(
+                                step_context=plan_context.for_step(
+                                    active_execution.get_step_by_key(key)
+                                ),
+                                step_failure_data=StepFailureData(
+                                    error=serializable_error, user_failure_data=None
+                                ),
+                            )
```

Comment:
not sure about failing here - technically this could fire after a STEP_SUCCESS event is already yielded right? (the same is theoretically possible with ChildProcessCrashException if it fires after the STEP_SUCCESS event)